app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -77,13 +77,15 @@ open class ProfilesViewModel(private val repository: ProfilesRepository) : ViewM
               profileList.filter { profile ->
 
                 // Filter by genders if specified
-                (genders == null || profile.gender in genders) &&
+                (genders == null || profile.gender in genders || genders.isEmpty()) &&
 
                     // Filter by age range if specified
                     (ageRange == null || profile.calculateAge() in ageRange) &&
 
                     // Filter by tags if specified
-                    (tags == null || profile.tags.containsAll(tags))
+                    (tags == null ||
```

Comment:
It seems more logical to me to display the profiles that have all the tags that the user requested.

Reply:
We discussed it a bit on the private discord, the problem is that the tagviewmodel was developed with "or" in the devs' mind, not an "and" so it would be very complicated for me to change that in the scope of this PR. Also as a team we haven't agreed on what's best as there are good points for both ideas. We'll see in the future if we change this